[bug] Failed to find_package() the "fmt" package installed by "MSVC x86 compiler".

### Problem Description

Recently, I used Conan to install `fmt` package with **"MSVC x86 compiler"**. However, there is an error when using `find_package()` to find "fmt" package.

```cmd
-- Conan: Component target declared 'fmt::fmt'
CMake Error at build/win32-msvc-x86-ninja-multi/conan-deps/cmakedeps_macros.cmake:39 (message):
  Library 'fmtd' not found in package.  If 'fmtd' is a system library,
  declare it with 'cpp_info.system_libs' property
Call Stack (most recent call first):
  build/win32-msvc-x86-ninja-multi/conan-deps/fmt-Target-debug.cmake:24 (conan_package_library_targets)
  build/win32-msvc-x86-ninja-multi/conan-deps/fmtTargets.cmake:26 (include)
  build/win32-msvc-x86-ninja-multi/conan-deps/fmt-config.cmake:10 (include)
  CMakeLists.txt:54 (find_package)


-- Configuring incomplete, errors occurred!
```

### Versions and Environments

* OS Version: **Windows 11**
* Compiler Version: **MSVC 16 (VS2019)**
  * MSVC toolset: **x86**
  * MSVC arch: **x86**
* CMake Version: **CMake 3.24.0**
* CMake Generator: **Ninja Multi-Config**
* Conan Version: **Conan 1.55.0**
* cmake-conan: [**0.19.0-dev**](https://github.com/conan-io/cmake-conan/blob/36ed835ac285aaf96ccd38bd0a3962fd59c12058/conan.cmake)

### Demo Project

[test-conan-cmake-profile.zip](https://github.com/conan-io/cmake-conan/files/10214406/test-conan-cmake-profile.zip)

* `CMakeLists.txt`

  <details><summary>Click to expand</summary>

  ```cmake
  cmake_minimum_required(VERSION 3.21)
  get_filename_component(folder_name "${CMAKE_CURRENT_SOURCE_DIR}" NAME)
  project(${folder_name} LANGUAGES C CXX)
  set(CMAKE_CONAN_OUTPUT_FOLDER "${PROJECT_BINARY_DIR}/conan-deps")
  include(conan.cmake)
  conan_cmake_profile(
      FILEPATH      "${CMAKE_CONAN_OUTPUT_FOLDER}/profile/build.ini"
      SETTINGS      os=Windows
                    arch=x86_64
                    build_type=Release
  )
  conan_cmake_profile(
      FILEPATH      "${CMAKE_CONAN_OUTPUT_FOLDER}/profile/host.ini"
      SETTINGS      os=Windows
                    arch=x86
                    build_type=Debug
                    compiler=msvc
                    compiler.version=192
                    compiler.runtime=dynamic
                    compiler.runtime_type=Debug
                    compiler.cppstd=14
      CONF          tools.cmake.build:jobs=10
                    tools.cmake.cmaketoolchain:generator="Ninja Multi-Config"
  )
  conan_cmake_install(
      PATH_OR_REFERENCE   "${PROJECT_SOURCE_DIR}/conanfile.py"
      INSTALL_FOLDER      "${CMAKE_CONAN_OUTPUT_FOLDER}"
      OUTPUT_FOLDER       "${CMAKE_CONAN_OUTPUT_FOLDER}"
      BUILD               missing
      REMOTE              conancenter
      PROFILE_BUILD       "${CMAKE_CONAN_OUTPUT_FOLDER}/profile/build.ini"
      PROFILE_HOST        "${CMAKE_CONAN_OUTPUT_FOLDER}/profile/host.ini"
  )
  list(APPEND CMAKE_MODULE_PATH "${CMAKE_CONAN_OUTPUT_FOLDER}")
  list(APPEND CMAKE_PREFIX_PATH "${CMAKE_CONAN_OUTPUT_FOLDER}")
  find_package(fmt REQUIRED)
  ```

  </details>

* `CMakePresets.json`

  <details><summary>Click to expand</summary>

  ```json
  {
    "version": 3,
    "cmakeMinimumRequired": {
      "major": 3,
      "minor": 21,
      "patch": 0
    },
    "configurePresets": [
      {
        "name": "win32-msvc-x86-ninja-multi",
        "displayName": "Windows MSVC x86 (Ninja Multi-Config)",
        "description": "Using MSVC x86 compiler with \"Ninja Multi-Config\" geneartor on Windows",
        "generator": "Ninja Multi-Config",
        "toolset": {
          "value": "v142,host=x86",
          "strategy": "external"
        },
        "architecture": {
          "value": "x86",
          "strategy": "external"
        },
        "binaryDir": "${sourceDir}/build/${presetName}",
        "installDir": "${sourceDir}/install/${presetName}",
        "cacheVariables": {
          "CMAKE_C_COMPILER": "cl.exe",
          "CMAKE_CXX_COMPILER": "cl.exe"
        }
      }
    ]
  }
  ```

  </details>

* `conanfile.py`

  <details><summary>Click to expand</summary>

  ```python
  from conans import ConanFile, CMake

  class ConsumerConan(ConanFile):
      settings = [ "os", "compiler", "build_type", "arch" ]
      generators = [
          "CMakeDeps", 
          "CMakeToolchain"
      ]
      requires = [
          "fmt/8.1.1@"
      ]
      default_options = {
          "fmt:shared": True
      }
  ```

  </details>

* [`conan.cmake`](https://github.com/conan-io/cmake-conan/blob/36ed835ac285aaf96ccd38bd0a3962fd59c12058/conan.cmake)

### Profiles

The following `build.ini` and `host.ini` profiles are generated by calling `conan_cmake_profile()`:

* `build.ini`

  ```ini
  [settings]
  os=Windows
  arch=x86_64
  build_type=Release
  ```

* `host.ini`

  ```ini
  [settings]
  os=Windows
  arch=x86
  build_type=Debug
  compiler=msvc
  compiler.version=192
  compiler.runtime=dynamic
  compiler.runtime_type=Debug
  compiler.cppstd=14
  [conf]
  tools.cmake.build:jobs=10
  tools.cmake.cmaketoolchain:generator="Ninja Multi-Config"
  ```

### Demonstration

1.  Download and unzip the above Demo Project.

2.  Remove the `fmt` package installed previously.

    ```cmd
    conan remove fmt/8.1.1@ --force
    ```

3.  Configure with `win32-msvc-x86-ninja-multi` preset:

    ```cmd
    vcvarsall.bat x86
    cmake --preset win32-msvc-x86-ninja-multi
    ```

### Logs

<details><summary>Click to expand</summary>

```cmd
F:\GitRepo\tmp\test-conan-cmake-profile>conan remove fmt/8.1.1@ --force

F:\GitRepo\tmp\test-conan-cmake-profile>vcvarsall.bat x86
**********************************************************************
** Visual Studio 2019 Developer Command Prompt v16.11.21
** Copyright (c) 2021 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x86'

F:\GitRepo\tmp\test-conan-cmake-profile>cmake --preset win32-msvc-x86-ninja-multi
Preset CMake variables:

  CMAKE_CXX_COMPILER="cl.exe"
  CMAKE_C_COMPILER="cl.exe"
  CMAKE_INSTALL_PREFIX:PATH="F:/GitRepo/tmp/test-conan-cmake-profile/install/win32-msvc-x86-ninja-multi"

-- The C compiler identification is MSVC 19.29.30147.0
-- The CXX compiler identification is MSVC 19.29.30147.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30133/bin/Hostx86/x86/cl.exe - skipped       
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30133/bin/Hostx86/x86/cl.exe - skipped     
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Conan: Creating profile F:/GitRepo/tmp/test-conan-cmake-profile/build/win32-msvc-x86-ninja-multi/conan-deps/profile/build.ini
-- Conan: Creating profile F:/GitRepo/tmp/test-conan-cmake-profile/build/win32-msvc-x86-ninja-multi/conan-deps/profile/host.ini
-- Conan: checking conan executable
-- Conan: Found program C:/Python/Python310/Scripts/conan.exe
-- Conan: Version found 1.55.0
-- Conan executing: C:/Python/Python310/Scripts/conan.exe install F:/GitRepo/tmp/test-conan-cmake-profile/conanfile.py --remote conancenter --install-folder F:/GitRepo/tmp/test-conan-cmake-profile/build/win32-msvc-x86-ninja-multi/conan-deps --output-folder F:/GitRepo/tmp/test-conan-cmake-profile/build/win32-msvc-x86-ninja-multi/conan-deps --build missing --profile:host F:/GitRepo/tmp/test-conan-cmake-profile/build/win32-msvc-x86-ninja-multi/conan-deps/profile/host.ini --profile:build F:/GitRepo/tmp/test-conan-cmake-profile/build/win32-msvc-x86-ninja-multi/conan-deps/profile/build.ini
Configuration (profile_host):
[settings]
arch=x86
build_type=Debug
compiler=msvc
compiler.cppstd=14
compiler.runtime=dynamic
compiler.runtime_type=Debug
compiler.version=192
os=Windows
[options]
[build_requires]
[env]
[conf]
tools.cmake.build:jobs=10
tools.cmake.cmaketoolchain:generator="Ninja Multi-Config"

Configuration (profile_build):
[settings]
arch=x86_64
build_type=Release
os=Windows
[options]
[build_requires]
[env]

fmt/8.1.1: Retrieving from server 'conancenter'
fmt/8.1.1: Trying with 'conancenter'...
Downloading conanmanifest.txt
Downloading conanfile.py
Downloading conan_export.tgz
fmt/8.1.1: Downloaded recipe revision 0
conanfile.py: Installing package
Requirements
    fmt/8.1.1 from 'conancenter' - Downloaded
Packages
    fmt/8.1.1:4375755290df55f8c855bcd5f433af4201eef7fc - Build

Cross-build from 'Windows:x86_64' to 'Windows:x86'
Installing (downloading, building) binaries...
fmt/8.1.1: Configuring sources in D:\.conan\short\5d5908\1\src
fmt/8.1.1: 
fmt/8.1.1: Copying sources to build folder
fmt/8.1.1: Building your package in D:\.conan\short\40c525\1
fmt/8.1.1: Generator txt created conanbuildinfo.txt
fmt/8.1.1: Calling generate()
fmt/8.1.1: Preset 'default' added to CMakePresets.json. Invoke it manually using 'cmake --preset default'
fmt/8.1.1: If your CMake version is not compatible with CMakePresets (<3.19) call cmake like: 'cmake <path> -G ""Ninja Multi-Config"" -DCMAKE_TOOLCHAIN_FILE=D:\.conan\short\40c525\1\build\generators\conan_toolchain.cmake -DFMT_DOC=OFF -DFMT_TEST=OFF -DFMT_INSTALL=ON -DFMT_LIB_DIR=lib -DFMT_OS=ON -DCMAKE_POLICY_DEFAULT_CMP0091=NEW'
fmt/8.1.1: Aggregating env generators
fmt/8.1.1: Calling build()
fmt/8.1.1: apply_conandata_patches(): No patches defined in conandata
fmt/8.1.1: CMake command: cmake -G ""Ninja Multi-Config"" -DCMAKE_TOOLCHAIN_FILE="D:/.conan/short/40c525/1/build/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="D:/.conan/short/2575db/1" -DFMT_DOC="OFF" -DFMT_TEST="OFF" -DFMT_INSTALL="ON" -DFMT_LIB_DIR="lib" -DFMT_OS="ON" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" "D:\.conan\short\40c525\1\src"
**********************************************************************
** Visual Studio 2019 Developer Command Prompt v16.11.21
** Copyright (c) 2021 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x64_x86'
CMake Warning:
  Ignoring extra path from command line:

   "D:/.conan/short/40c525/1/build/Multi-Config"


-- CMake version: 3.24.0
-- Using Conan toolchain: D:/.conan/short/40c525/1/build/generators/conan_toolchain.cmake
-- Conan toolchain: C++ Standard 14 with extensions OFF
-- Conan toolchain: Setting BUILD_SHARED_LIBS = ON
-- The CXX compiler identification is MSVC 19.29.30147.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30133/bin/Hostx86/x86/cl.exe - skipped     
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Module support is disabled.
-- Version: 8.1.1
-- Build type: Release
-- CXX_STANDARD: 14
-- Performing Test has_std_14_flag
-- Performing Test has_std_14_flag - Failed
-- Performing Test has_std_1y_flag
-- Performing Test has_std_1y_flag - Failed
-- Performing Test SUPPORTS_USER_DEFINED_LITERALS
-- Performing Test SUPPORTS_USER_DEFINED_LITERALS - Success
-- Performing Test FMT_HAS_VARIANT
-- Performing Test FMT_HAS_VARIANT - Success
-- Required features: cxx_variadic_templates
-- Looking for _strtod_l
-- Looking for _strtod_l - found
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_POLICY_DEFAULT_CMP0091


-- Build files have been written to: D:/.conan/short/40c525/1/build
fmt/8.1.1: CMake command: cmake --build "D:\.conan\short\40c525\1\build" --config Debug -- -j8
**********************************************************************
** Visual Studio 2019 Developer Command Prompt v16.11.21
** Copyright (c) 2021 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x64_x86'
[1/3] Building CXX object CMakeFiles\fmt.dir\src\os.cc.obj
[2/3] Building CXX object CMakeFiles\fmt.dir\src\format.cc.obj
[3/3] Linking CXX shared library bin\fmt.dll
fmt/8.1.1: Package '4375755290df55f8c855bcd5f433af4201eef7fc' built
fmt/8.1.1: Build folder D:\.conan\short\40c525\1\build
fmt/8.1.1: Generated conaninfo.txt
fmt/8.1.1: Generated conanbuildinfo.txt
fmt/8.1.1: Generating the package
fmt/8.1.1: Package folder D:\.conan\short\2575db\1
fmt/8.1.1: Calling package()
fmt/8.1.1: Copied 1 '.rst' file: LICENSE.rst
fmt/8.1.1: CMake command: cmake --install "D:\.conan\short\40c525\1\build" --config Debug --prefix "D:/.conan/short/2575db/1"
**********************************************************************
** Visual Studio 2019 Developer Command Prompt v16.11.21
** Copyright (c) 2021 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x64_x86'
-- Installing: D:/.conan/short/2575db/1/lib/fmt.lib
-- Installing: D:/.conan/short/2575db/1/bin/fmt.dll
-- Installing: D:/.conan/short/2575db/1/lib/cmake/fmt/fmt-config.cmake
-- Installing: D:/.conan/short/2575db/1/lib/cmake/fmt/fmt-config-version.cmake
-- Installing: D:/.conan/short/2575db/1/lib/cmake/fmt/fmt-targets.cmake
-- Installing: D:/.conan/short/2575db/1/include/fmt/args.h
-- Installing: D:/.conan/short/2575db/1/include/fmt/chrono.h
-- Installing: D:/.conan/short/2575db/1/include/fmt/color.h
-- Installing: D:/.conan/short/2575db/1/include/fmt/compile.h
-- Installing: D:/.conan/short/2575db/1/include/fmt/core.h
-- Installing: D:/.conan/short/2575db/1/include/fmt/format.h
-- Installing: D:/.conan/short/2575db/1/include/fmt/format-inl.h
-- Installing: D:/.conan/short/2575db/1/include/fmt/locale.h
-- Installing: D:/.conan/short/2575db/1/include/fmt/os.h
-- Installing: D:/.conan/short/2575db/1/include/fmt/ostream.h
-- Installing: D:/.conan/short/2575db/1/include/fmt/printf.h
-- Installing: D:/.conan/short/2575db/1/include/fmt/ranges.h
-- Installing: D:/.conan/short/2575db/1/include/fmt/xchar.h
-- Installing: D:/.conan/short/2575db/1/lib/pkgconfig/fmt.pc
fmt/8.1.1 package(): Packaged 1 '.dll' file: fmt.dll
fmt/8.1.1 package(): Packaged 13 '.h' files
fmt/8.1.1 package(): Packaged 1 '.lib' file: fmt.lib
fmt/8.1.1 package(): Packaged 1 '.rst' file: LICENSE.rst
fmt/8.1.1: Package '4375755290df55f8c855bcd5f433af4201eef7fc' created
fmt/8.1.1: Created package revision f4ed97587d584702f74dfe6b577168ce
conanfile.py: Generator txt created conanbuildinfo.txt
conanfile.py: Generator 'CMakeDeps' calling 'generate()'
conanfile.py: Generator 'CMakeToolchain' calling 'generate()'
conanfile.py: Preset 'default' added to CMakePresets.json. Invoke it manually using 'cmake --preset default'
conanfile.py: If your CMake version is not compatible with CMakePresets (<3.19) call cmake like: 'cmake <path> -G ""Ninja Multi-Config"" -DCMAKE_TOOLCHAIN_FILE=F:\GitRepo\tmp\test-conan-cmake-profile\build\win32-msvc-x86-ninja-multi\conan-deps\conan_toolchain.cmake -DCMAKE_POLICY_DEFAULT_CMP0091=NEW'
conanfile.py: Aggregating env generators
conanfile.py: Generated conaninfo.txt
conanfile.py: Generated graphinfo
-- Conan: Component target declared 'fmt::fmt'
CMake Error at build/win32-msvc-x86-ninja-multi/conan-deps/cmakedeps_macros.cmake:39 (message):
  Library 'fmtd' not found in package.  If 'fmtd' is a system library,
  declare it with 'cpp_info.system_libs' property
Call Stack (most recent call first):
  build/win32-msvc-x86-ninja-multi/conan-deps/fmt-Target-debug.cmake:24 (conan_package_library_targets)
  build/win32-msvc-x86-ninja-multi/conan-deps/fmtTargets.cmake:26 (include)
  build/win32-msvc-x86-ninja-multi/conan-deps/fmt-config.cmake:10 (include)
  CMakeLists.txt:52 (find_package)


-- Configuring incomplete, errors occurred!
See also "F:/GitRepo/tmp/test-conan-cmake-profile/build/win32-msvc-x86-ninja-multi/CMakeFiles/CMakeOutput.log".
```

</details>

--- a/README.md
+++ b/README.md
@@ -484,8 +484,8 @@ conan_cmake_profile(FILEPATH      "${CMAKE_BINARY_DIR}/profile"
                                   compiler.cppstd=14
                     OPTIONS       fmt:shared=True
                                   fmt:header_only=False
-                    CONF          tools.cmake.cmaketoolchain:generator="Visual Studio 16 2019"
-                                  tools.cmake.cmaketoolchain:toolset_arch=x64
+                    CONF          "tools.cmake.cmaketoolchain:generator=Visual Studio 16 2019"
+                                  "tools.cmake.cmaketoolchain:toolset_arch=x64"
                     TOOL_REQUIRES cmake/3.16.3)
 ```
 

--- a/tests.py
+++ b/tests.py
@@ -896,9 +896,9 @@ class CMakeConanTest(unittest.TestCase):
                               compiler.cppstd=14
                 OPTIONS       fmt:shared=True
                               fmt:header_only=False
-                CONF          tools.cmake.cmaketoolchain:generator="Visual Studio 16 2019"
-                              tools.cmake.cmaketoolchain:toolset_arch=x64
-                              tools.cmake.build:jobs=10
+                CONF          "tools.cmake.cmaketoolchain:generator=Visual Studio 16 2019"
+                              "tools.cmake.cmaketoolchain:toolset_arch=x64"
+                              "tools.cmake.build:jobs=10"
                 ENV           "MyPath1=(path)/some/path11"
                               "MyPath1=+(path)/other/path12"
                 BUILDENV      "MyPath2=(path)/some/path21"
@@ -922,7 +922,7 @@ class CMakeConanTest(unittest.TestCase):
             fmt:shared=True
             fmt:header_only=False
             [conf]
-            tools.cmake.cmaketoolchain:generator="Visual Studio 16 2019"
+            tools.cmake.cmaketoolchain:generator=Visual Studio 16 2019
             tools.cmake.cmaketoolchain:toolset_arch=x64
             tools.cmake.build:jobs=10
             [env]

